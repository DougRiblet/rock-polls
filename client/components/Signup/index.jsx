import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { signUpUser } from '../../actions/actions';
import signupPresent from './signup-presentational';

const mapStateToProps = state => ({
  authenticated: state.authenticated,
  user: state.user,
});

const mapDispatchToProps = dispatch => bindActionCreators({ signUpUser }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(signupPresent);

