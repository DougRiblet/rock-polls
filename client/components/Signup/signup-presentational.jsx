// @flow

import React from 'react';

type Props = {
  signUpUser: (string, string) => mixed,
  authenticated: boolean,
  user_id: string,
  username: string,
};

type State = {
  username: string,
  password: string,
};

export default class Signup extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      username: '',
      password: '',
    };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChangeUsername = this.handleChangeUsername.bind(this);
    this.handleChangePassword = this.handleChangePassword.bind(this);
  }

  handleSubmit: Function;
  handleChangeUsername: Function;
  handleChangePassword: Function;

  // eslint-disable-next-line no-undef
  handleChangeUsername(event: SyntheticInputEvent<*>) {
    this.setState({ username: event.target.value });
  }

  // eslint-disable-next-line no-undef
  handleChangePassword(event: SyntheticInputEvent<*>) {
    this.setState({ password: event.target.value });
  }

  // eslint-disable-next-line no-undef
  handleSubmit(event: SyntheticInputEvent<*>) {
    event.preventDefault();
    this.props.signUpUser(this.state.username, this.state.password);
    this.setState({ username: '', password: '' });
  }

  render() {
    return (
      <div id='signup'>
        <h3>Sign Up</h3>
        <form onSubmit={this.handleSubmit}>
          <label>
            Username:
            <input
              id='username-input'
              type='text'
              value={this.state.username}
              minLength='4'
              maxLength='15'
              size='20'
              onChange={this.handleChangeUsername}
            />
          </label>
          <br />
          <label>
          Password:
            <input
              id='password-input'
              type='text'
              value={this.state.password}
              minLength='6'
              maxLength='19'
              size='20'
              onChange={this.handleChangePassword}
            />
          </label>
          <br />
          <input type='submit' value='submit' />
        </form>
        <div className='show-status'>
          <p>Authenticated: {String(this.props.authenticated)}</p>
          <p>Username: {this.props.username}</p>
          <p>User ID: {this.props.user_id}</p>
        </div>
      </div>

    );
  }
}
