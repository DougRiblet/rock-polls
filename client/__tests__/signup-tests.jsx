import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Signup from '../components/Signup/signup-presentational';

describe('Signup', () => {
  test('Snapshot', () => {
    const Signup_tree = renderer.create(<Signup
      signUpUser={() => signUpUser()}
      authenticated={false}
      user_id={''}
      username={''}
    />).toJSON();
    expect(Signup_tree).toMatchSnapshot();
  });

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Signup
      signUpUser={() => signUpUser()}
      authenticated={false}
      user_id={''}
      username={''}
    />, div);
  });
});
