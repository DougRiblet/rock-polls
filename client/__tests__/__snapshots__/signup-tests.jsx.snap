// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Snapshot 1`] = `
<div
  id="signup"
>
  <h3>
    Sign Up
  </h3>
  <form
    onSubmit={[Function]}
  >
    <label>
      Username:
      <input
        id="username-input"
        maxLength="15"
        minLength="4"
        onChange={[Function]}
        size="20"
        type="text"
        value=""
      />
    </label>
    <br />
    <label>
      Password:
      <input
        id="password-input"
        maxLength="19"
        minLength="6"
        onChange={[Function]}
        size="20"
        type="text"
        value=""
      />
    </label>
    <br />
    <input
      type="submit"
      value="submit"
    />
  </form>
  <div
    className="show-status"
  >
    <p>
      Authenticated: 
      false
    </p>
    <p>
      Username: 
      
    </p>
    <p>
      User ID: 
      
    </p>
  </div>
</div>
`;
